src/app/quiz/page.tsx

```diff
@@ -82,43 +158,124 @@ export default function Quiz() {
+              {selectedTopic === 0 ? (
                 <p className="flex align-middle justify-center font-semibold text-3xl">
                   Please Select a Topic{" "}
                 </p>
               ) : (
                 <p className="flex align-middle justify-center font-semibold text-3xl">
+                  Click the button below to Generate questions for lecture{" "}
                   {selectedTopic}
                 </p>
               )}
 
-              <div className="flex justify-center align-middle">
-                <button
-                  className="bg-blue-500 text-white p-4 m-4 rounded-lg"
-                  disabled={selectedTopic === ""}
-                  onClick={() => generateQuiz(selectedTopic)}
-                >
-                  Generate quiz!
-                </button>
-              </div>
+              {isLoading ? (
+                <div className="flex justify-center mt-4">
+                  <p className="text-xl font-semibold">Loading...</p>
+                </div>
+              ) : (
+                <div className="flex justify-center align-middle">
+                  <button
+                    className="bg-blue-500 text-white p-4 m-4 rounded-lg"
+                    disabled={selectedTopic === 0}
+                    onClick={() => generateQuiz(selectedTopic)}
+                  >
+                    Generate quiz!
+                  </button>
+                </div>
+              )}
             </div>
           ) : (
-            <div className="w-full h-full rounded-xl border p-4 border-gray-400 bg-white shadow-lg">
-              <p>Something Went Wrong!</p>
+            <div className="w-full h-full rounded-xl border p-4 border-gray-400 bg-white shadow-lg overflow-auto">
+              {response.map((question, index) => (
+                <div key={question.id} className="mb-4 border-b-2 p-2 py-4">
+                  <p className="font-semibold pb-2">
+                    {index + 1}. {question.question}
+                  </p>
+                  <div className="flex flex-col pl-6">
+                    {(
+                      ["option_a", "option_b", "option_c"] as Array<
+                        keyof QuestionData
+                      >
+                    ).map((optionKey) => (
+                      <div key={optionKey}>
+                        <input
+                          id={`${question.id}_${optionKey}`}
+                          name={`question_${question.id}`}
+                          value={optionKey}
+                          type="radio"
+                          className="mr-3"
+                          onChange={(e) =>
+                            handleAnswerChange(
+                              question.question_number,
+                              e.target.value
+                            )
+                          }
+                        />
+                        <span className="mr-2">{question[optionKey]}</span>
```

Comment:
Could here be added the question alphabet? Like "a. This is the question"

Reply:
okay I can do that